src/app/shared/wordlist-generator/wordlist-generator.component.html

```diff
@@ -0,0 +1,104 @@
+          </button>
+        </div>
+
+        <div *ngFor="let control of names.controls.slice(1); let i = index">
+          <mat-form-field>
+            <mat-label>Name {{ i + 2 }}</mat-label>
+            <input matInput [formControl]="getFormControl(control)" placeholder="Enter name {{ i + 2 }}"/>
+          </mat-form-field>
+
+          <button mat-icon-button color="warn" type="button" (click)="removeItem(i + 1, 'names')" *ngIf="names.controls.length > 1">
+            <mat-icon>delete</mat-icon>
+          </button>
+        </div>
+      </div>
+
+      <!-- Display added special dates -->
+      <div formArrayName="sparetext">
+        <div fxLayout="row" fxLayoutGap="10px">
+          <mat-form-field>
+            <mat-label>Special Attribute 1</mat-label>
+            <input matInput [formControl]="getFormControl(sparetext.controls[0])" placeholder="Enter name"/>
+            <mat-icon matSuffix color="primary" matTooltip="Target's {{ specialhint }}">help</mat-icon>
+          </mat-form-field>
+
+          <button mat-icon-button color="primary" type="button" (click)="addControl('sparetext')">
+            <mat-icon>add</mat-icon>
+          </button>
+        </div>
+
+        <div *ngFor="let control of sparetext.controls.slice(1); let i = index">
+          <mat-form-field>
+            <mat-label>Special Attribute {{ i + 2 }}</mat-label>
+            <input matInput [formControl]="getFormControl(control)" placeholder="Enter Special Attribute {{ i + 2 }}"/>
+          </mat-form-field>
+
+          <button mat-icon-button color="warn" type="button" (click)="removeItem(i + 1, 'sparetext')" *ngIf="sparetext.controls.length > 1">
+            <mat-icon>delete</mat-icon>
+          </button>
+        </div>
+      </div>
+
+      <div>
+        <mat-form-field>
+          <mat-label>Special Date</mat-label>
+          <input matInput [matDatepicker]="datepicker" formControlName="specialdates" name="specialdate" placeholder="Enter special date"/>
+          <mat-datepicker-toggle matSuffix [for]="datepicker"></mat-datepicker-toggle>
+          <mat-datepicker #datepicker></mat-datepicker>
+        </mat-form-field>
+      </div>
+
+      </div>
+
+      <!-- Additional Options -->
+      <mat-accordion>
+        <mat-expansion-panel>
+          <mat-expansion-panel-header>
+            <mat-panel-title>
+              <span><b>Additional Options</b></span>
+            </mat-panel-title>
+          </mat-expansion-panel-header>
+          <div fxLayout="column" fxLayoutAlign="start stretch" fxLayoutGap="10px">
+            <mat-form-field>
+              <mat-label>File Name</mat-label>
+              <input matInput formControlName="filename" name="filename" placeholder="Wordlist.txt" />
+              <mat-hint align="start">
+                {{ filenamehint }}
+              </mat-hint>
+            </mat-form-field>
+          </div>
+          <mat-checkbox formControlName="useSpecCharacters">Add special chars</mat-checkbox>
```

Comment:
Maybe we should have text like these as variables instead? That way it is easy to implement support for other languages in the future.

Reply:
Changed to variables